src/main/minio.js

```diff
@@ -1471,79 +1473,40 @@ export class Client {
     this.makeRequest({method, bucketName, query}, policyPayload, 204, '', cb)
   }
 
-  // Generate a presigned URL for PUT. Using this URL, the browser can upload to S3 only with the specified object name.
-  //
-  // __Arguments__
-  // * `bucketName` _string_: name of the bucket
-  // * `objectName` _string_: name of the object
-  // * `expiry` _number_: expiry in seconds (optional, default 7 days)
-  presignedPutObject(bucketName, objectName, expires, cb) {
-    if (this.anonymous) {
-      throw new errors.AnonymousRequestError('Presigned PUT url cannot be generated for anonymous requests')
-    }
-    if (isFunction(expires)) {
-      cb = expires
-      expires = 24 * 60 * 60 * 7
-    }
-    if (!isValidBucketName(bucketName)) {
-      throw new errors.InvalidBucketNameError('Invalid bucket name: ' + bucketName)
-    }
-    if (!isValidObjectName(objectName)) {
-      throw new errors.InvalidObjectNameError(`Invalid object name: ${objectName}`)
-    }
-    if (!isNumber(expires)) {
-      throw new TypeError('expires should be of type "number"')
-    }
-    var method = 'PUT'
-    var requestDate = new Date()
-    this.getBucketRegion(bucketName, (e, region) => {
-      if (e) return cb(e)
-      // This statement is added to ensure that we send error through
-      // callback on presign failure.
-      var url
-      var reqOptions = this.getRequestOptions({method,
-                                               region,
-                                               bucketName,
-                                               objectName})
-      try {
-        url = presignSignatureV4(reqOptions, this.accessKey, this.secretKey,
-                                 region, requestDate, expires)
-      } catch (pe) {
-        return cb(pe)
-      }
-      cb(null, url)
-    })
-  }
-
-  // Generate a presigned URL for GET
+  // Generate a generic presigned URL which can be
+  // used for HTTP methods GET, PUT, HEAD and DELETE
   //
   // __Arguments__
+  // * `method` _string_: name of the HTTP method
   // * `bucketName` _string_: name of the bucket
   // * `objectName` _string_: name of the object
   // * `expiry` _number_: expiry in seconds (optional, default 7 days)
   // * `respHeaders` _object_: response headers to override (optional)
-  presignedGetObject(bucketName, objectName, expires, respHeaders, cb) {
+  generatePresignedUrl(method, bucketName, objectName, expires, reqParams, respHeaders, cb) {
```

Comment:
since `respHeaders` is specific to `GET` request, you can keep it only in `presignedGet`

generatePresignedUrl can look like:
generatePresignedUrl(method, bucketName, objectName, expires, reqParams, cb)

i.e if you want to call generatePresignedUrl() from presignedGet() you can use reqParams to pass respHeaders

Reply:
The name of the function generatePresignedUrl does not align with the names presignedGet presignedPut. we can call it just presignedUrl() instead. We should change the name in other libs also to follow same naming conventions.

Reply:
Done.